nifi-toolkit/nifi-property-encryptor-tool/src/main/java/org/apache/nifi/PropertyEncryptorCommand.java

```diff
@@ -0,0 +1,281 @@
+                        flowEncryptor.processFlow(inputStream, outputStream, inputEncryptor, outputEncryptor);
+                    }
+                } catch (IOException e) {
+                    logger.error("Failed to encrypt flow definition file: [{}]", flow.getAbsolutePath(), e);
+                }
+            }
+        }
+    }
+
+    public void outputSensitiveProperties(final PropertyEncryptionMethod algorithm, final String sensitivePropertyKey) {
+        final File outputPropertiesFile = ConfigurationFileUtils.getOutputFile(outputDirectory, applicationPropertiesFile);
+        final MutableApplicationProperties updatedProperties = new MutableApplicationProperties(new Properties());
+        updatedProperties.setProperty(NiFiProperties.SENSITIVE_PROPS_KEY, sensitivePropertyKey);
+        updatedProperties.setProperty(NiFiProperties.SENSITIVE_PROPS_ALGORITHM, algorithm.name());
+
+        try {
+            try (InputStream inputStream = new FileInputStream(applicationPropertiesFile);
+                 FileOutputStream outputStream = new FileOutputStream(outputPropertiesFile)) {
+                new StandardPropertiesWriter().writePropertiesFile(inputStream, outputStream, updatedProperties);
+            }
+            logger.info("Output sensitive properties key to {}", applicationPropertiesFile.getAbsolutePath());
+        } catch (IOException e) {
+            logger.error("Failed to output sensitive property keys after encrypting flow", e);
+        }
+    }
+
+    public void outputKeyToBootstrap() throws IOException {
+        final File bootstrapFile = bootstrapLoader.getBootstrapFileWithinConfDirectory(confDirectory);
+        final File outputBootstrapFile = ConfigurationFileUtils.getOutputFile(outputDirectory, bootstrapFile);
+        final MutableBootstrapProperties bootstrapProperties = bootstrapLoader.loadMutableBootstrapProperties(bootstrapFile.getPath());
+        bootstrapProperties.setProperty(BootstrapProperties.BootstrapPropertyKey.SENSITIVE_KEY.getKey(), outputHexKey);
+        try (InputStream inputStream = new FileInputStream(bootstrapFile);
+             FileOutputStream outputStream = new FileOutputStream(outputBootstrapFile)) {
+            new StandardPropertiesWriter().writePropertiesFile(inputStream, outputStream, bootstrapProperties);
+            logger.info("Output the bootstrap key to {}", outputBootstrapFile);
+        }
+    }
+
+    private XmlEncryptor getXmlEncryptor(final ProtectionScheme scheme) {
+        final SensitivePropertyProviderFactory providerFactory = StandardSensitivePropertyProviderFactory.withKey(inputHexKey);
+        return new XmlEncryptor(providerFactory, scheme);
+    }
+
+    private XmlDecryptor getXmlDecryptor(final SensitivePropertyProviderFactory providerFactory, final ProtectionScheme scheme) {
+        return new XmlDecryptor(providerFactory, scheme);
+    }
+
+    private String getExistingRootKey(final Path confDirectory, final String passphrase) {
```

Comment:
It looks like this is supposed to fall back to the supplied passphrase, but the `passphrase` argument is not specified.